db/views/materialized_patient_summaries_v03.sql

```diff
@@ -0,0 +1,394 @@
+with latest_bp_passport as (
+    select distinct on (patient_id) *
+    from patient_business_identifiers
+    where identifier_type = 'simple_bp_passport' and deleted_at is null
+    order by patient_id, device_created_at desc
+), latest_phone_number as (
+    select distinct on (patient_id) *
+    from patient_phone_numbers
+    where deleted_at is null
+    order by patient_id, device_created_at desc
+), latest_medical_history as (
+    SELECT DISTINCT ON (patient_id) * FROM medical_histories WHERE deleted_at IS NULL
+),ranked_prescription_drugs as (
+    select
+        bp.id as bp_id,
+        array_agg(array[name, dosage] order by prescription_drugs.is_protocol_drug desc, prescription_drugs.name, prescription_drugs.device_created_at desc) as blood_pressure_drugs,
```

Comment:
Using `group by` here instead of using rank. 

The earlier version had to join onto itself 5 times, a ranked_prescription_drugs result set of cardinality `O(no. of bps * no. of prescription drugs)`. The current version simply returns a result set of cardinality of `O(no. of bps)` with all the prescription drugs in the correct order in the array_agg.